homeassistant/components/daikin/config_flow.py

```diff
@@ -128,7 +129,7 @@ async def async_step_discovery(self, discovery_info):
     async def async_step_zeroconf(self, discovery_info):
         """Prepare configuration for a discovered Daikin device."""
         _LOGGER.debug("Zeroconf discovery_info: %s", discovery_info)
-        await self.async_set_unique_id(discovery_info[KEY_HOSTNAME])
+        await self.async_set_unique_id(get_mac_address(ip=discovery_info[CONF_HOST]))
```

Comment:
Please note that getting the mac address might introduce new and unexpected behavior, as you are not guaranteed to get the mac address of the end device (might also be a device in between, in those cases, thus you might get the same mac address for 2 different devices).

Reply:
But (correct me if I'm wrong) Zeroconf only works on the local network. And the mac-address might reflect a router, if the host is on a different (layer2) network.

So this is good enough (much better than the previous solution).

Reply:
I think the real solution is down below 😉

Reply:
Oh btw, zeroconf/mdns is not limited to the local network. This will break on my network for sure, as I have mDNS repeaters set up between the VLANs for example.

Reply:
Ok!

Reply:
> I think the real solution is down below 😉

I wish that I had 5-10 different models (as they all seem to do things slightly different) but I only have one and it doesn't support Zeroconf at all..

I'll use the pydaikin method of getting mac (via UDP), using the code in https://bitbucket.org/mustang51/pydaikin/src/master/pydaikin/discovery.py.